### PR TITLE
created new template cisco ios show environment temperature plus raw …

### PR DESCRIPTION
**Pull request by socretz**
…and parsed tested

##### ISSUE TYPE

 - New Template Pull Request
 - Bugfix Pull Request
 - Additional Testing
 - Docs Pull Request

##### SUMMARY

```

```

### CODE REVIEW

**Comment by socretz** [comment]
New at contributing. 

New template Cisco iOS show environment temperature

File: templates/cisco_ios_show_environment_temperature.template [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Value INLET_TEMPERATURE_STATE (\w+)
+Value INLET_YELLOW_THRESHOLD (\d+)
+Value INLET_RED_THRESHOLD (\d+)
+Value HOTSPOT_TEMPERATURE_VALUE (\d+)
+Value HOTSPOT_TEMPERATURE_STATE (\w+)
+Value HOTSPOT_YELLOW_THRESHOLD (\d+)
+Value HOTSPOT_RED_THRESHOLD (\d+)
+
+Start
+  ^Switch\s+${SWITCH}:\s+SYSTEM\s+TEMPERATURE\s+is\s+${SWITCH_TEMPERATURE}\s*$$
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
  ^Switch\s+\d -> Continue.Record
  ^Switch\s+${SWITCH}:\s+SYSTEM\s+TEMPERATURE\s+is\s+${SWITCH_TEMPERATURE}\s*$$
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 2]
```diff
@@ -0,0 +1,41 @@
+Value SWITCH (\d+)
+Value SWITCH_TEMPERATURE (\w+)
+Value INLET_TEMPERATURE_VALUE (\d+)
+Value INLET_TEMPERATURE_STATE (\w+)
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
Value INLET_TEMPERATURE_STATE (\S+)
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+Value INLET_YELLOW_THRESHOLD (\d+)
+Value INLET_RED_THRESHOLD (\d+)
+Value HOTSPOT_TEMPERATURE_VALUE (\d+)
+Value HOTSPOT_TEMPERATURE_STATE (\w+)
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
Value HOTSPOT_TEMPERATURE_STATE (\S+)
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+Value HOTSPOT_YELLOW_THRESHOLD (\d+)
+Value HOTSPOT_RED_THRESHOLD (\d+)
+
+Start
+  ^Switch\s+${SWITCH}:\s+SYSTEM\s+TEMPERATURE\s+is\s+${SWITCH_TEMPERATURE}\s*$$
+  ^Inlet\s+Temperature\s+Value:\s+${INLET_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$ -> Inlet
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
  ^Inlet\s+Temperature\s+Value:\s+${INLET_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$ -> Inlet
  ^\s*$$
  ^. -> Error
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+ 
+Inlet
+  ^Temperature\s+State:\s+${INLET_TEMPERATURE_STATE}\s*$$
+  ^Yellow\s+Threshold\s+:\s+${INLET_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
+  ^Red\s+Threshold\s+:\s+${INLET_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Hotspot
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
  ^Switch\s+\d -> Continue.Record
  ^Switch\s+${SWITCH}:\s+SYSTEM\s+TEMPERATURE\s+is\s+${SWITCH_TEMPERATURE}\s*$$ -> Start
  ^\s*$$
  ^. -> Error

```

File: templates/cisco_ios_show_environment_temperature.template [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  
+  
+  
+  
+  
+  
+
+  
+
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  
+  
+  
+  
+  
+  
+
+  
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+  
+  
+  
+  
+  
+  
+
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+
+  
+  
+  
+  
+  
+  
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+
+
+  
+  
+  
+  
+  
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+
+
+
+  
+  
+  
+  
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+ 
+
+
+
+
+
+  
+  
+  
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Record Start
+ 
+
+
+
+
+
+  
+  
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+  ^Yellow\s+Threshold\s+:\s+${HOTSPOT_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
+  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Record Start
+ 
+
+
+
+
+
+  
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+  ^Temperature\s+State:\s+${HOTSPOT_TEMPERATURE_STATE}\s*$$
+  ^Yellow\s+Threshold\s+:\s+${HOTSPOT_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
+  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Record Start
+ 
+
+
+
+
+
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+  ^Hotspot\s+Temperature\s+Value:\s+${HOTSPOT_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$
+  ^Temperature\s+State:\s+${HOTSPOT_TEMPERATURE_STATE}\s*$$
+  ^Yellow\s+Threshold\s+:\s+${HOTSPOT_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
+  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Record Start
+ 
+
+
+
+
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+Hotspot
+  ^Hotspot\s+Temperature\s+Value:\s+${HOTSPOT_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$
+  ^Temperature\s+State:\s+${HOTSPOT_TEMPERATURE_STATE}\s*$$
+  ^Yellow\s+Threshold\s+:\s+${HOTSPOT_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
+  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Record Start
+ 
+
+
+
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+ 
+Hotspot
+  ^Hotspot\s+Temperature\s+Value:\s+${HOTSPOT_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$
+  ^Temperature\s+State:\s+${HOTSPOT_TEMPERATURE_STATE}\s*$$
+  ^Yellow\s+Threshold\s+:\s+${HOTSPOT_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
+  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Record Start
+ 
+
+
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+  ^Red\s+Threshold\s+:\s+${INLET_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Hotspot
+ 
+Hotspot
+  ^Hotspot\s+Temperature\s+Value:\s+${HOTSPOT_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$
+  ^Temperature\s+State:\s+${HOTSPOT_TEMPERATURE_STATE}\s*$$
+  ^Yellow\s+Threshold\s+:\s+${HOTSPOT_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
+  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Record Start
+ 
+
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+  ^Yellow\s+Threshold\s+:\s+${INLET_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
+  ^Red\s+Threshold\s+:\s+${INLET_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Hotspot
+ 
+Hotspot
+  ^Hotspot\s+Temperature\s+Value:\s+${HOTSPOT_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$
+  ^Temperature\s+State:\s+${HOTSPOT_TEMPERATURE_STATE}\s*$$
+  ^Yellow\s+Threshold\s+:\s+${HOTSPOT_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
+  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Record Start
+ 
+
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
```

File: templates/cisco_ios_show_environment_temperature.template [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+Inlet
+  ^Temperature\s+State:\s+${INLET_TEMPERATURE_STATE}\s*$$
+  ^Yellow\s+Threshold\s+:\s+${INLET_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
+  ^Red\s+Threshold\s+:\s+${INLET_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Hotspot
+ 
+Hotspot
+  ^Hotspot\s+Temperature\s+Value:\s+${HOTSPOT_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$
+  ^Temperature\s+State:\s+${HOTSPOT_TEMPERATURE_STATE}\s*$$
+  ^Yellow\s+Threshold\s+:\s+${HOTSPOT_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
+  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Record Start
```
Comment (jmcgill298) [inline templates/cisco_ios_show_environment_temperature.template]:
```suggestion
  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$
  ^Switch\s+\d -> Continue.Record
  ^Switch\s+${SWITCH}:\s+SYSTEM\s+TEMPERATURE\s+is\s+${SWITCH_TEMPERATURE}\s*$$ - > Start
  ^\s*$$
  ^. -> Error
```

File: tox.ini [hunk 22]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py36
+envlist = py37
```
Comment (jmcgill298) [inline tox.ini]:
```suggestion
envlist = py36
```

File: tests/cisco_ios/show_environment_temperature/cisco_ios_show_environment_temperature.raw [hunk 23]
```diff
@@ -0,0 +1,11 @@
...
+Inlet Temperature Value: 28 Degree Celsius
+Temperature State: GREEN
+Yellow Threshold : 46 Degree Celsius
+Red Threshold    : 56 Degree Celsius
+
+Hotspot Temperature Value: 68 Degree Celsius
+Temperature State: GREEN
+Yellow Threshold : 105 Degree Celsius
+Red Threshold    : 125 Degree Celsius
+
```
Comment (jmcgill298) [inline tests/cisco_ios/show_environment_temperature/cisco_ios_show_environment_temperature.raw]:
```suggestion

Switch 2: SYSTEM TEMPERATURE is OK
Inlet Temperature Value: 28 Degree Celsius
Temperature State: GREEN
Yellow Threshold : 46 Degree Celsius
Red Threshold    : 56 Degree Celsius

Hotspot Temperature Value: 68 Degree Celsius
Temperature State: GREEN
Yellow Threshold : 105 Degree Celsius
Red Threshold    : 125 Degree Celsius
```

File: tests/cisco_ios/show_environment_temperature/cisco_ios_show_environment_temperature.parsed [hunk 24]
```diff
@@ -0,0 +1,13 @@
...
+-  switch: '1'
+   switch_temperature: OK
+   inlet_temperature_value: '28'
+   inlet_temperature_state: GREEN
+   inlet_yellow_threshold: '46'
+   inlet_red_threshold: '56'
+   hotspot_temperature_value: '68'
+   hotspot_temperature_state: GREEN
+   hotspot_yellow_threshold: '105'
+   hotspot_red_threshold: '125'
```
Comment (jmcgill298) [inline tests/cisco_ios/show_environment_temperature/cisco_ios_show_environment_temperature.parsed]:
```suggestion
   hotspot_red_threshold: '125'
 -  switch: '2'
   switch_temperature: OK
   inlet_temperature_value: '28'
   inlet_temperature_state: GREEN
   inlet_yellow_threshold: '46'
   inlet_red_threshold: '56'
   hotspot_temperature_value: '68'
   hotspot_temperature_state: GREEN
   hotspot_yellow_threshold: '105'
   hotspot_red_threshold: '125'
```

File: templates/cisco_ios_show_environment_temperature.template (hunk 1 above)
Comment (socretz) [reply templates/cisco_ios_show_environment_temperature.template]: Made changes.  Works.  Moving to next suggestion.

File: templates/cisco_ios_show_environment_temperature.template (hunk 2 above)
Comment (socretz) [reply templates/cisco_ios_show_environment_temperature.template]: tested and works.  Moving to next suggestion

File: templates/cisco_ios_show_environment_temperature.template (hunk 3 above)
Comment (socretz) [reply templates/cisco_ios_show_environment_temperature.template]: Test moving to next

File: templates/cisco_ios_show_environment_temperature.template (hunk 4 above)
Comment (socretz) [reply templates/cisco_ios_show_environment_temperature.template]: The error statement actually threw an error when evaluating Inlet Temperature Value. The value is repeated as a hotspot so I moved the line to Inlet and added error checking after each section... hard to explain see below:

Original
Start
^Switch\s+\d -> Continue.Record
^Switch\s+${SWITCH}:\s+SYSTEM\s+TEMPERATURE\s+is\s+${SWITCH_TEMPERATURE}\s*$
^Inlet\s+Temperature\s+Value:\s+${INLET_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$ 
$ -> Inlet
  ^\s*$$
  ^. -> Error 
 
Inlet
  ^Temperature\s+State:\s+${INLET_TEMPERATURE_STATE}\s*$$
  ^Yellow\s+Threshold\s+:\s+${INLET_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
  ^Red\s+Threshold\s+:\s+${INLET_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Hotspot
  ^\s*$$
  ^. -> Error 
 
Hotspot
  ^Hotspot\s+Temperature\s+Value:\s+${HOTSPOT_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$
  ^Temperature\s+State:\s+${HOTSPOT_TEMPERATURE_STATE}\s*$$
  ^Yellow\s+Threshold\s+:\s+${HOTSPOT_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Record Start
  ^\s*$$
  ^. -> Error 


Now
Start
  ^Switch\s+\d -> Continue.Record
  ^Switch\s+${SWITCH}:\s+SYSTEM\s+TEMPERATURE\s+is\s+${SWITCH_TEMPERATURE}\s*$$ -> Inlet
  ^\s*$$
  ^. -> Error 
 
Inlet
  ^Inlet\s+Temperature\s+Value:\s+${INLET_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$ 
  ^Temperature\s+State:\s+${INLET_TEMPERATURE_STATE}\s*$$
  ^Yellow\s+Threshold\s+:\s+${INLET_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
  ^Red\s+Threshold\s+:\s+${INLET_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Hotspot
  ^\s*$$
  ^. -> Error 
 
Hotspot
  ^Hotspot\s+Temperature\s+Value:\s+${HOTSPOT_TEMPERATURE_VALUE}\s+Degree\s+Celsius\s*$$
  ^Temperature\s+State:\s+${HOTSPOT_TEMPERATURE_STATE}\s*$$
  ^Yellow\s+Threshold\s+:\s+${HOTSPOT_YELLOW_THRESHOLD}\s+Degree\s+Celsius\s*$$
  ^Red\s+Threshold\s+:\s+${HOTSPOT_RED_THRESHOLD}\s+Degree\s+Celsius\s*$$ -> Record Start
  ^\s*$$
  ^. -> Error

File: templates/cisco_ios_show_environment_temperature.template (hunk 21 above)
Comment (socretz) [reply templates/cisco_ios_show_environment_temperature.template]: same as previous

File: tox.ini (hunk 22 above)
Comment (socretz) [reply tox.ini]: will make sure on next pull request

File: tests/cisco_ios/show_environment_temperature/cisco_ios_show_environment_temperature.parsed (hunk 24 above)
Comment (socretz) [reply tests/cisco_ios/show_environment_temperature/cisco_ios_show_environment_temperature.parsed]: added second switch.  Will see on next pull request

**Comment by socretz** [comment]
I am closing this pull request and will open a new one with the new changes.  Thank you for the input.  Greatly appreciated examples.

**Pull request closed without merging**